packages/shared/lib/services/config.service.ts

```diff
@@ -74,6 +75,51 @@ class ConfigService {
         return encryptionManager.decryptProviderConfig(result);
     }
 
+    validateProviderConfig(authMode: AuthModeType, providerConfig: ProviderConfig): string[] {
+        const missingFields = [];
+
+        switch (authMode) {
+            case 'OAUTH1':
+            case 'OAUTH2':
+            case 'TBA':
+                if (providerConfig.oauth_client_id === '') {
+                    missingFields.push('oauth_client_id');
+                }
+
+                if (providerConfig.oauth_client_secret === '') {
+                    missingFields.push('oauth_client_secret');
+                }
+                break;
+
+            case 'APP':
+                if (providerConfig.oauth_client_id === '') {
+                    missingFields.push('oauth_client_id');
+                }
+
+                if (providerConfig.oauth_client_secret === '') {
+                    missingFields.push('oauth_client_secret');
+                }
+
+                if (providerConfig.app_link === '') {
+                    missingFields.push('app_link');
+                }
+                break;
+
+            case 'CUSTOM':
+                break;
+
+            case 'BASIC':
+            case 'API_KEY':
+            case 'APP_STORE':
+            case 'TABLEAU':
+            case 'NONE':
+            case 'OAUTH2_CC':
+                break;
+        }
+
+        return missingFields;
+    }
+
```

Comment:
what about a more functional approach that is more extensible. @bodinsamuel is not gonna like it 😆 
```typescript
    interface ValidationRule {
        field: keyof ProviderConfig;
        modes: AuthModeType[];
        validate: (value?: ProviderConfig[keyof ProviderConfig]) => boolean;
    }

    isEmpty = (value?: ProviderConfig[keyof ProviderConfig]) => { ... }

    VALIDATION_RULES: ValidationRule[] = [
        {
            field: 'oauth_client_id',
            modes: ['OAUTH1', 'OAUTH2', 'TBA', 'APP'],
            validate: isEmpty
        },
        {
            field: 'oauth_client_secret',
            modes: ['OAUTH1', 'OAUTH2', 'TBA', 'APP'],
            validate: isEmpty
        },
        {
            field: 'app_link',
            modes: ['APP'],
            validate: isEmpty
        }
    ];

    validateProviderConfig(authMode: AuthModeType, providerConfig: ProviderConfig): string[] {
        return this.VALIDATION_RULES
            .filter((rule) => rule.modes.includes(authMode))
            .filter((rule) => !rule.validate(providerConfig[rule.field]))
            .map((field) => field.toString());
    }

```